pkg/controller/service_lb.go

```diff
@@ -0,0 +1,392 @@
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels:      labels,
+					Annotations: podAnnotations,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            "lb-svc",
+							Image:           image,
+							Command:         []string{"bash"},
+							Args:            []string{"-c", "while true; do sleep 10000; done"},
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							SecurityContext: &corev1.SecurityContext{
+								Privileged:               &privileged,
+								AllowPrivilegeEscalation: &allowPrivilegeEscalation,
+							},
+						},
+					},
+				},
+			},
+			Strategy: v1.DeploymentStrategy{
+				Type: v1.RecreateDeploymentStrategyType,
+			},
+		},
+	}
+	return
+}
+
+func (c *Controller) updateLbSvcDeployment(svc *corev1.Service, dp *v1.Deployment) *v1.Deployment {
+	attachmentName, attachmentNs, _ := c.checkAttachNetwork(svc)
+	providerName := getAttachNetworkProvider(svc)
+	attchSubnetAnnotation := fmt.Sprintf(util.LogicalSwitchAnnotationTemplate, providerName)
+	attchIpAnnotation := fmt.Sprintf(util.IpAddressAnnotationTemplate, providerName)
+	podAnnotations := map[string]string{
+		util.AttachmentNetworkAnnotation: fmt.Sprintf("%s/%s", attachmentNs, attachmentName),
+		attchSubnetAnnotation:            svc.Annotations[attchSubnetAnnotation],
+	}
+	if svc.Spec.LoadBalancerIP != "" {
+		podAnnotations[attchIpAnnotation] = svc.Spec.LoadBalancerIP
+	}
+	dp.Spec.Template.Annotations = podAnnotations
+
+	return dp
+}
+
+func (c *Controller) createLbSvcPod(svc *corev1.Service) error {
+	var deploy *v1.Deployment
+	var err error
+	needToCreate := false
+	if deploy, err = c.config.KubeClient.AppsV1().Deployments(svc.Namespace).Get(context.Background(), genLbSvcDpName(svc.Name), metav1.GetOptions{}); err != nil {
+		if k8serrors.IsNotFound(err) {
+			needToCreate = true
+		} else {
+			return err
+		}
+	}
+
+	if needToCreate {
+		newDp := c.genLbSvcDeployment(svc)
+		_, err := c.config.KubeClient.AppsV1().Deployments(svc.Namespace).
+			Create(context.Background(), newDp, metav1.CreateOptions{})
+
+		if err != nil {
+			klog.Errorf("failed to create deployment %s, err: %v", newDp.Name, err)
+			return err
+		}
+		return nil
+	} else {
+		deploy = c.updateLbSvcDeployment(svc, deploy)
+		_, err := c.config.KubeClient.AppsV1().Deployments(svc.Namespace).Update(context.Background(), deploy, metav1.UpdateOptions{})
+		if err != nil {
+			klog.Errorf("failed to update deployment %s, err: %v", deploy.Name, err)
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (c *Controller) getLbSvcPod(svcName, svcNamespace string) (*corev1.Pod, error) {
+	sel, _ := metav1.LabelSelectorAsSelector(&metav1.LabelSelector{
+		MatchLabels: map[string]string{"app": genLbSvcDpName(svcName), "namespace": svcNamespace},
+	})
+
+	pods, err := c.podsLister.Pods(svcNamespace).List(sel)
+	if err != nil {
```

Comment:
should loop here to wait ready

Reply:
the loop exists where the function is called